woocommerce/woocommerce-ios
# Login: Support setting up Jetpack connection for logging in with site credentials

Part of #7597
⚠️  Please make sure to also review https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/678 ⚠️


### Description

When a user tries to log in with the site credentials for an account that does not have a WP.com account connected to the site's Jetpack, we display the account mismatch screen. This experience can be improved by supporting them log in to a WP.com account to connect with Jetpack. 

This PR improves the experience using the Jetpack REST API set up in #7600. Changes on the UI layer include:
- Added a new view model for the Jetpack connection error screen.
- Added a new view model for the web view that handles Jetpack connection.
- Added an extension to `WordPressOrgCredentials` to create a `CookieNonceAuthenticator` based on the credentials information.
- Updated `AuthenticationManager` to check for WP.org credentials to detect when there's an issue with Jetpack connection. 

Analytics will be handled in a subsequent PR since this PR is already too big.

### Testing instructions

- Create a JN site with Jetpack and WooCommerce.
- Invite a  test user to the site.
- Open the app and log out if needed.
- On the prologue screen, select the "Enter your site address" button.
- Enter the address of the test site.
- Proceed to select "Sign in with site credentials"
- Enter the credentials of the test account.
- After the login succeeds, notice that a new error screen is displayed.
- Select the "Connect Jetpack" button.
- Notice that a web view is presented and you are asked to log in to a WP.com account.
- After the login, proceed to authorize the Jetpack connection.
- After the authorization completes, notice that a new screen is presented asking to verify the authorized WP.com email address. Enter the password or use the magic link flow to proceed.
- After the login succeeds, you should be navigated to the home screen .

### Screenshots


https://user-images.githubusercontent.com/5533851/189015317-73883353-f048-48aa-aacc-a37db370949d.mp4





---
